Finalize removal of typed pointers

dwightguth opened the pull request:
Previously we still had some references to typed pointers in .ll files and in API calls that still created typed pointers when working with older versions of LLVM. This led to some invalid bitcode being generated in ways that wasn't necessarily a problem, but which caused problems when I was making changes to that code. This PR is a preliminary to further changes which removes all remaining references to the API constructor for typed pointers and removes all typed pointers from flat bitcode files.

Robertorosmaninho reviewed (changes requested):
I made some comments to improve the code quality. Sorry for being pedantic.
Also, `runtime/main/main.ll` still contains typed pointers, can this be a problem in the future?

dwightguth commented on lib/codegen/EmitConfigParser.cpp:
```diff
@@ -574,8 +574,7 @@ static void emit_get_token(kore_definition *definition, llvm::Module *module) {
           module, "init_float", llvm::Type::getVoidTy(ctx),
-          llvm::PointerType::getUnqual(float_ty),
-          llvm::PointerType::getUnqual(ctx));
+          llvm::PointerType::getUnqual(ctx), llvm::PointerType::getUnqual(ctx));
```
Yes; I made the changes completely procedurally, so cleanups like this didn't get done, but I will go ahead and handle it in places that I notice it being called in more than one place in a function.

Robertorosmaninho replied:
Thank you! And sorry again for being pedantic!

dwightguth commented:
Don't worry at all; your suggestions do all make the code much nicer; I don't mind doing it and appreciate the thorough review. I believe I have now addressed all the issues. I will mark you for re-review as soon as I confirm that the tests are still passing.

theo25 reviewed (approved):
Looks good to me

Robertorosmaninho reviewed (approved):
Thank you for addressing all comments! The code looks much nicer indeed!
LGTM!
